lighthouse-core/computed/metrics/cumulative-layout-shift-all-frames.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * @license Copyright 2020 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const makeComputedArtifact = require('../computed-artifact.js');
+
+class CumulativeLayoutShiftAllFrames {
+  /**
+   * @param {LH.Trace} trace
+   * @return {Promise<{value: number}>}
+   */
+  static async compute_(trace) {
+    const cumulativeShift = trace.traceEvents
+      .filter(e => e.name === 'LayoutShift' && e.args && e.args.data && e.args.data.score)
```

Comment:
You can narrow the type by annotating the filter function to be a type guard. here's an example: https://github.com/GoogleChrome/lighthouse/pull/7817/files#diff-b66c6ac9b4b8c0726c736eb7479a9e4ff22c7b81bfa1b2b259c2cdfa220ea1f8R19

Reply:
oh nvm the type here is kinda complicated, a ts ignore on L19 is probably better

Reply:
totes forgot about the recent input thing. that was a rough bug